templates/webapps/galaxy/visualization/gie.mako

```diff
@@ -0,0 +1,157 @@
+
+    ${h.js('libs/jquery/select2')}
+
+    <div style="overflow: auto; height: 100%;">
+        <div class="page-container" style="padding: 10px;">
+            <div class="col-md-8 col-md-offset 2 col-xs-12">
+                <div class="row">
+                    <h1>Galaxy Interactive Environment Launcher</h1>
+                    <p>
+                        Galaxy Interactive Environments (GIEs) launch embedded,
+                        dockerized versions of popular data analysis suites such as
+                        Jupyter and RStudio, right from within Galaxy. They allow you
+                        to dynamically interact with your data, right on the server. No
+                        more uploading and downloading between various platforms just
+                        to get your work done!
+                        <br />
+                        <a href="https://docs.galaxyproject.org/en/master/admin/interactive_environments.html">Admin Docs</a>
+                    </p>
+                    <form id='launcher' action="NONE" method="GET">
+
+                        <table class="table table-striped">
+                            <tr>
+                                <td>GIE: </td>
+                                <td>
+                                    <span id="image_name" style="width: 400px" />
+                                </td>
+                            </tr>
+                            <tr>
+                                <td>Image: </td>
+                                <td>
+                                    <span id="image_tag" style="width:400px" />
+                                    <input id="image_tag_hidden" type="hidden" name="image_tag" value="NONE" />
+                                    <p id="image_desc">
+                                    </p>
+                                </td>
+                            </tr>
+                            <tr>
+                                <td>Datasets: </td>
+                                <td>
+                                    <span id="additional_datasets" style="width:400px"></span>
+                                    <input id="dataset_id" name="dataset_id" type="hidden">
+                                    <input id="additional_dataset_ids" name="additional_dataset_ids" type="hidden">
+                                </td>
+                            </tr>
+                        </table>
+                        <input type="submit" class="button" value="Launch" disabled="">
+                    </form>
+                </div>
+            </div>
+
+
+        </div>
+    </div>
+
+<script type="text/javascript">
+$(document).ready(function(){
+    var gie_image_map = {
+        % for image_name in gie_image_map.keys():
+            "${image_name}": [
+                % for image in gie_image_map[image_name]:
+                {
+                    id: "${image['image']}",
+                    text: "${image['image']}",
```

Comment:
![](https://cloud.githubusercontent.com/assets/458683/12012248/270f8a26-acb2-11e5-9007-82c4169fa978.png) Is ^ not enough? what do you have in mind?